esp/src/src-react/components/forms/CheckPermissions.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from "react";
+import { DefaultButton, Dropdown, IDropdownOption, MessageBar, MessageBarType, PrimaryButton, TextField } from "@fluentui/react";
+import { scopedLogger } from "@hpcc-js/util";
+import { useForm, Controller } from "react-hook-form";
+import nlsHPCC from "src/nlsHPCC";
+import { MessageBox } from "../../layouts/MessageBox";
+import { FilePermission } from "src/ws_access";
+
+const logger = scopedLogger("src-react/components/forms/CheckPermissions.tsx");
+
+interface CheckPermissionsFormValues {
+    FileName: string;
+    SelectedValue: string;
+}
+
+const defaultValues: CheckPermissionsFormValues = {
+    FileName: "",
+    SelectedValue: "",
+};
+
+interface CheckPermissionsFormProps {
+    refreshGrid?: () => void;
+    showForm: boolean;
+    setShowForm: (_: boolean) => void;
+}
+
+export const CheckPermissionsForm: React.FunctionComponent<CheckPermissionsFormProps> = ({
+    refreshGrid,
+    showForm,
+    setShowForm
+}) => {
+
+    const { handleSubmit, control, reset, setValue, watch } = useForm<CheckPermissionsFormValues>({ defaultValues });
+
+    const [showError, setShowError] = React.useState(false);
+    const [errorMessage, setErrorMessage] = React.useState("");
+    const [userOptions, setUserOptions] = React.useState<IDropdownOption[]>([]);
+    const [groupOptions, setGroupOptions] = React.useState<IDropdownOption[]>([]);
+    const [filePermissionResponse, setFilePermissionResponse] = React.useState<string>("");
+
+    React.useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const { FilePermissionResponse } = await FilePermission({});
+                const { Users, Groups } = FilePermissionResponse;
+                const sortedUsers = Users?.User?.sort((a, b) => a.username.localeCompare(b.username));
+                const sortedGroups = Groups?.Group?.sort((a, b) => a.name.localeCompare(b.name));
+
+                setUserOptions(sortedUsers?.map((user: { username: any; }) => {
+                    return { key: user.username, text: user.username };
+                }) ?? []);
+
+                setGroupOptions(sortedGroups?.map(group => {
+                    return { key: group.name, text: group.name };
+                }) ?? []);
+            } catch (error) {
+                logger.error(error);
+            }
+        };
+
+        fetchData();
+    }, []);
+
+    const closeForm = React.useCallback(() => {
+        reset(defaultValues);
+        setShowForm(false);
+        setFilePermissionResponse("");
+    }, [reset, setShowForm]);
+
+    const onSubmit = async (data: CheckPermissionsFormValues) => {
+        try {
+            const { FileName, SelectedValue } = data;
+            logger.info(`Checking permissions for file ${FileName} for selected value: ${SelectedValue}`);
+            const response = await FilePermission({
+                request: {
```

Comment:
This is why SelectedValue should be separated into two. The FilePermission response is not reliable if both UserName and GroupName are submitted with the same value.